File: DemoMethods/Indexes/OrderByCompanyAndCountry.cs
using System.Linq;
using DemoMethods.Entities;
using Raven.Client.Indexes;

namespace DemoMethods.Indexes
{
    public class OrderByCompanyAndCountry : AbstractIndexCreationTask<Order>
    {
        public OrderByCompanyAndCountry()
        {
            Map = orders => from order in orders
                            select new
                            {
                                ShipTo_City = order.ShipTo.City,
                                ShipTo_Country = order.ShipTo.Country
                            };
        }
    }
}

File: DemoMethods/src/main/java/net/ravendb/demo/indexes/OrderByCompanyAndCountry.java
package net.ravendb.demo.indexes;

import net.ravendb.client.indexes.AbstractIndexCreationTask;

public class OrderByCompanyAndCountry extends AbstractIndexCreationTask {

    public OrderByCompanyAndCountry() {
        map = "from order in docs.Orders " +
                "select new { " +
                "   ShipTo_City = order.ShipTo.City, " +
                "   ShipTo_Country = order.ShipTo.Country" +
                "}";
    }
}
